src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String SAVE_QUERY = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_QUERY = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_QUERY = "SELECT * FROM books";
+    private static final String UPDATE_QUERY = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID_QUERY = "DELETE FROM books WHERE id = ?";
+    private static final String ZERO_INSERTED_ROWS_MESSAGE
+            = "Expected to insert at least 1 row, but inserted 0 rows";
+    private static final String CANNOT_SAVE_BOOK_MESSAGE = "Cannot save this book: ";
+    private static final String CANNOT_FIND_BOOK_BY_ID_MESSAGE = "Cannot find a book by this id: ";
+    private static final String CANNOT_FIND_BOOKS_MESSAGE = "Cannot find books from the table";
+    private static final String ZERO_UPDATED_BOOKS_MESSAGE = "No books were updated in the table";
+    private static final String CANNOT_UPDATE_BOOK_MESSAGE = "Cannot update this book: ";
```

Comment:
It's a not completed message, but rather message template, because you are adding id of a book to it, so constant name could be a bit confusing.